src/Sql/SqlBase.php

```diff
@@ -277,7 +312,12 @@ public function alwaysQuery($query, $input_file = null, $result_file = '')
         // We show the query when --debug is used and this function created the temp file.
         $this->logQueryInDebugMode($query, $input_file_original);
 
-        $success = drush_always_exec($exec);
+        // This Process deliberately ignores simulated mode.
```

Comment:
Maybe we could just create a Symfony Process object directly if we don't want to respect simulated mode? That would also bypass logging, which may or may not be an issue.

Reply:
Yeah, I think the logging is useful, so I'm inclined to use our Process object.